BaseClasses.py

```diff
@@ -1041,6 +1041,7 @@ class Location:
     shop_slot_disabled: bool = False
     event: bool = False
     locked: bool = False
+    hidden = False
```

Comment:
To someone reading the code, it's not clear what "hidden" means.
"Does it mean this location is difficult to find? Does it mean the item at this location is invisible?"

You can put documentation on an attribute by putting a triple-quote string right below it.
```python3
    locked: bool = False
    hidden = False
    """ `hidden` means ... """
    game: str = "Generic"
```

Also, do we want type annotations on everything here? (My personal style would be that no annotation is needed if there's a clear type in the default value, but it looks like someone would like types on everything here.)